src/PaymentGateways/Actions/RegisterPaymentGatewayRoutes.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Give\PaymentGateways\Actions;
+
+use Give\Framework\PaymentGateways\Routes\ReturnFromOffsiteRedirectRoute;
+use Give\Helpers\Hooks;
+
+class RegisterPaymentGatewayRoutes
```

Comment:
Rather than setting up one action per route (n+1), use a single action which matches the query string with the listener.

Reply:
Fixed this up to be more clear / fluent 8ad87141